cdap-ui/app/hydrator/services/hydrator-node-service.js

```diff
@@ -22,42 +22,51 @@ class HydratorPlusPlusNodeService {
     this.IMPLICIT_SCHEMA = IMPLICIT_SCHEMA;
     this.GLOBALS = GLOBALS;
   }
-  getPluginInfo(node, appType, sourceConn, artifactVersion) {
+  getPluginInfo(node, appType, sourceConnections, sourceNodes, artifactVersion) {
     var promise;
     if (angular.isObject(node._backendProperties) && Object.keys(node._backendProperties).length) {
       promise = this.$q.when(node);
     } else {
       promise = this.HydratorPlusPlusHydratorService.fetchBackendProperties(node, appType, artifactVersion);
     }
-    return promise.then((node) => this.configurePluginInfo(node, appType, sourceConn));
+    return promise.then((node) => this.configurePluginInfo(node, appType, sourceConnections, sourceNodes));
   }
-  configurePluginInfo(node, appType, sourceConn) {
-    const getSchema = (node, targetNode) => {
-      var schema = node.outputSchema;
+  configurePluginInfo(node, appType, sourceConnections, sourceNodes) {
+    const getInputSchema = (sourceNode, targetNode, sourceConnections) => {
+      let schema = '';
+      let isStreamSource = sourceNode.plugin.name === 'Stream';
+      let inputSchema;
 
-      if (targetNode.type === 'errortransform' && this.GLOBALS.pluginConvert[node.type] !== 'source') {
-        if (Array.isArray(node.inputSchema) && !node.inputSchema.length) {
-          return null;
-        }
-        schema = node.inputSchema && Array.isArray(node.inputSchema) ? node.inputSchema[0].schema : node.inputSchema;
-      } else if (targetNode.type === 'errortransform' && this.GLOBALS.pluginConvert[node.type] === 'source') {
-        return null;
-      }
-
-      var isStreamSource = false;
-      var inputSchema;
       const isFieldExists = (field, schema) => {
         if(angular.isObject(schema) && Array.isArray(schema.fields)) {
           return schema.fields.filter(schemaField => schemaField.name === field.name).length;
         }
         return false;
       };
-      if (node.plugin.name === 'Stream') {
-        isStreamSource = true;
+
+      if (typeof sourceNode.outputSchema === 'string') {
+        sourceNode.outputSchema = [this.getOutputSchemaObj(sourceNode.outputSchema)];
+      }
+
+      if (sourceNode.outputSchema[0].name !== this.GLOBALS.defaultSchemaName) {
```

Comment:
This will throw an error if the source node hasn't been opened and doesn't have a schema?